### PR TITLE
#189 - add paperless-ng template

### LINKED ISSUE #189: CA Template Request - paperless-ng
## Paperless-ng - a (beautiful) paperless fork

* Github: https://github.com/jonaswinkler/paperless-ng
* Docker Hub: jonaswinkler/paperless-ng
* Documentation: https://paperless-ng.readthedocs.io/en/latest/
* Migration from paperless to paperless-ng: https://paperless-ng.readthedocs.io/en/latest/setup.html?highlight=migration#migration-to-paperless-ng

Paperless-ng is a well-maintained fork of [paperless](https://github.com/selfhosters/unRAID-CA-templates/blob/master/templates/paperless.xml). Unfortunately, the later hasn't received a lot of updates and bug fixes in the past. Even pull requests are not merged for some time now. Thus, I propose to add paperless-ng as **additional** alternative to paperless and papermerge to the Unraid CA. 

Here are a few key features of paperless-ng:

* New front end build with Angular. It features full text search with scored and highlighted results, savable filters, a dashboard, and document uploading on the landing page.
* Mobile support is also almost there. Some layouts don't work yet on small screens.
* New mail consumer that supports multiple accounts and custom filters and actions. Fully tested!
* Paperless-ng trains a neural network on your documents and assigns tags and corerspondents automatically, if you instruct it to do so.
* Updated dependencies.
* More tests of critical backend parts.
* A proper task processing queue that can consume multiple documents in parallel. Consumption of many documents is now blazing fast on multi core system. Fixed much of the consumer code, so that it does not block the database during consumption, for instance.
* Compatible with the paperless iOS and Android [app](https://github.com/bauerj/paperless_app)

Paperless-ng supports SQlite and PostgresQL as database. I suggest to stick with SQlite as the default configuration. Paperless-ng is available as pre-release right now. So, we might want to flag the template with beta in its name.

What do you guys think about paperless-ng?

Read [here](https://github.com/the-paperless-project/paperless/issues/711) for more information about the difference between paperless and paperless-ng.

### PR DESCRIPTION
**Pull request by Tooa**
Closes #189

### CODE REVIEW

**Comment by jrdnlc** [comment]
Please change branch to testing.

**Comment by Tooa** [comment]
@jrdnlc swiched to `testing`. Please do not merge yet. For some reason uploaded documents are not processed in the current state. Will have to investigate more.

**Comment by jonaswinkler** [comment]
Awesome stuff. How exactly does this work? Will it pull images from docker hub or build the image from the GitHub repository? Never used unRAID.

Ping me if anything is required for this.

**Comment by Roxedus** [comment]
@jonaswinkler It will grab it from dockerhub, the template is used to form a docker run command

**Comment by Tooa** [comment]
I think I'm finished. Can someone take a look at this?

**Comment by jrdnlc** [comment]
I can merge it and test.

File: templates/paperless-ng.xml
Comment (Roxedus) [inline templates/paperless-ng.xml:15]: Might be worth converting this to just use markdown syntax, as that is the way forward.

**Review by Roxedus — CHANGES_REQUESTED** [review]
Good job! Some small things I would do differently

**Comment by Tooa** [comment]
Thanks for your review @Roxedus. I like your suggestions! Will tackle your comments soon later the day.

File: templates/paperless-ng.xml
Comment (Tooa) [reply templates/paperless-ng.xml:26]: What about replacing the `localhost` with `<ip>`? In addition, I might add the redis requirement to the overview or description section of the document.

File: templates/paperless-ng.xml
Comment (Tooa) [reply templates/paperless-ng.xml:15]: Sounds good. Is there any reference how this markdown syntax in the templates look like? Couldn't find examples in the repository ether.

File: templates/paperless-ng.xml
Comment (Roxedus) [reply templates/paperless-ng.xml:26]: That was mainly the change I was fishing for.

File: templates/paperless-ng.xml
Comment (Tooa) [reply templates/paperless-ng.xml:15]: Also, what is the difference between the `Overview` and `Description` tag?

Comment (Roxedus) [reply templates/paperless-ng.xml:15]: Not really, it is a thing with the 6.9. I will look into updating the others later. 
```md
# This is a test

| markdown | table |
| --- | --- |
| yes | no
```
gives 
image
So its not fully implemented.

Comment (Roxedus) [reply templates/paperless-ng.xml:15]: > Also, what is the difference between the `Overview` and `Description` tag?

Overview gets used, unlike description

File: templates/paperless-ng.xml
Comment (Tooa) [reply templates/paperless-ng.xml:26]: Okay, I will do so then. But need to replace the `<ip>` with `[ip]` as `<>` are reserved chars in XML.

File: templates/paperless-ng.xml
Comment (Tooa) [reply templates/paperless-ng.xml]: I played around a little bit with the markdown syntax in the template and I'm not happy with it. I suggest to postpone this change until 6.9 is out. I'm willing to change this template myself then. For now, let's stick to the current state.

**Review by Roxedus — APPROVED** [review]

**Comment by jonaswinkler** [comment]
Awesome. If anyone asks me again how to setup this on unRaid, I'll simply point them towards here.

**Comment by Tooa** [comment]
@jonaswinkler Once @Roxedus merges the `testing` branch into `master`, the template will be available in Unraids community application store (see [here](https://forums.unraid.net/topic/38582-plug-in-community-applications/)). As maintainer of the paperless Unraid template (see [here](https://forums.unraid.net/topic/87196-support-paperless-docker/)), I will try to get a few users into testing `paperless-ng` in order to generate feedback for you.

**Pull request merged**
